app/src/androidTest/java/nl/rug/www/rugsummerschools/controller/LoginActivityTest.java

```diff
@@ -0,0 +1,112 @@
+package nl.rug.www.rugsummerschools.controller;
+
+import android.content.Context;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+
+import nl.rug.www.rugsummerschools.R;
+
+import static android.support.test.InstrumentationRegistry.getTargetContext;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.clearText;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.action.ViewActions.typeText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.RootMatchers.withDecorView;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static nl.rug.www.rugsummerschools.controller.DrawableMatcher.withDrawable;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.core.IsNot.not;
+
+@RunWith(AndroidJUnit4.class)
+public class LoginActivityTest {
+
+    /**
+     * Passcode must be 8 characters long and contain only lower letters and digits
+     */
+    private static final String DEFAULT_PASSCODE = "11111111";
+    private static final String WRONG_PASSCODE = "11111112";
+    private static final String SHORT_PASSCODE = "abcdefg";
+    private static final String LONG_PASSCODE = "1o12jij1j21i2ji1ij3";
+    private static final String NOT_ALLOWED_PASSCODE = "AB+-aiso";
+
+    @Rule
+    public ActivityTestRule<LoginActivity> mActivityTestRule = new ActivityTestRule<>(LoginActivity.class, false, false);
```

Comment:
The book that I read says that the convention style of android is to put m in front of a member. I added m as a prefix on AndroidStudio so that it does the job for me. The book is 3rd edition Android Programming the big nerd ranch guide by Bill Phillips.

Reply:
good. 
i seem to have read somewhere that this was obsolete, 
but it's a long time since i did Java, and I don't know the 
conventions anymore. if this is the Android convention, t
then keep it!